COMPILING.md

```diff
@@ -0,0 +1,139 @@
+
+Note that you don’t need to install the JACK package(s) for a headless build. If you plan to run headless on Gentoo, or are compiling under Ubuntu for use on another Ubuntu machine, the only packages you should need for a headless build are `qtcore`, `qtnetwork`, `qtconcurrent` and `qtxml` (both for building and running the server).
+
+Compile the sources to ignore the JACK sound library:
+
+```
+make distclean ;# reccommended
+qmake "CONFIG+=nosound headless"
+make
+sudo make install ;# optional
+```
+
+To control the server with systemd, see this [unit file example](https://github.com/jamulussoftware/jamulus/blob/master/distributions/jamulus-server.service). See also runtime [configuration options](/wiki/Command-Line-Options), and [this information](/wiki/Tips-Tricks-More#controlling-recording-on-linux-headless-servers) on controlling recordings on headless servers.
+
+---
+
+## Windows
+
+
+You will need [Qt](https://www.qt.io/download)
+
+* Use the free GPLv2 license for Open Source development
+* To determine the Qt version you need, check [qt-installer-windows.qs](https://github.com/jamulussoftware/jamulus/blob/master/windows/qt-installer-windows.qs): under INSTALL_COMPONENTS you will see `qt.qt5.[version]`, e.g., 5123 means version 5.12.3.
+* Select Components during installation: Expand the Qt section, find the matching version, e.g., **Qt 5.12.3**, and add the compiler components for your compiler, e.g., `MSVC 2017 32-bit/64-bit` for Visual Studio 2019
+* [ASIO development files](https://www.steinberg.net/en/company/developer.html)
+
+### Compiling and building installer
+
+Most users will probably want to use this method:
+
+1. Open PowerShell
+1. Navigate to the `jamulus` directory
+1. To allow unsigned scripts, right-click on the `windows\deploy_windows.ps1` script, select properties and allow the execution of this script. You can also run `Set-ExecutionPolicy -ExecutionPolicy RemoteSigned -Scope CurrentUser`. (You can revert this after having run this script. For more information see the [Microsoft PowerShell documentation page](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.security/set-executionpolicy)).
+1. Edit the $QtCompile32 and $QtCompile64 variables.
+1. Run the Jamulus compilation and installer script in PowerShell: `.\windows\deploy_windows.ps1`.
+1. You can now find the Jamulus installer in the `.\deploy` directory.
+
+### Compiling only
+
+1. Create a folder under `\windows` called ASIOSDK2
+1. Download the [ASIOSDK](https://www.steinberg.net/asiosdk), open the top level folder in the .zip file and copy the contents into `[\path\to\jamulus\source]\windows\ASIOSDK2` so that, e.g., the folder `[\path\to\jamulus\source]\windows\ASIOSDK2\common` exists
+1. Open Jamulus.pro in Qt Creator, configure the project with a default kit, then compile & run
+
+---
+
+## macOS
```

Comment:
Have you checked this section against the Github automation scripts?

Reply:
No, but can/should we just refer people to the build scripts for all this stuff? I'd rather not maintain stuff in two places. Or is it not as simple as that?

Reply:
That's the top level.

In fact, for all platforms for which we _have_ autobuilds, `COMPILING.md` should be redundant.  (I could fork and generate builds on github whenever I want using the autobuilds, if I was a new developer, for example.)  We should make it quite clear for those platforms that the autobuild process is the preferred method of generating the binaries.

It should really only be for platforms we do not do autobuilds for that this document becomes the normal process.

And ideally, as we become aware of demand on a platform, we should introduce autobuild support (subject to Github-compatible provision being available).

Reply:
> We should make it quite clear for those platforms that the autobuild process is the preferred method of generating the binaries.

So is a just a one-liner at the start? _"Please note that for supported platforms, the autobuild process in the Jamulus Github repository is the preferred method of generating binaries"_?

Reply:
More that "the binaries generated by the autobuild process in the Jamulus repository are the preferred method of supporting these platforms".  There should be no benefit doing it any other way.  If there is, we should get to hear about it and see whether the autobuild process needs changing.

Reply:
Oh OK, so in effect saying that unless you're compiling for an unsupported platform, you're better off using the binaries instead of compiling. Does that need saying? I suppose it can't do any harm :-)